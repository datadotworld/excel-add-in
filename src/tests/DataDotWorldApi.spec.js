import papa from 'papaparse';

const testData = [
  [1, 2, 3], // normal data
  [1, '2, 3, 4', 5], // data with commas
  ['10/12/2012', '17%', 0.123], // mixed data
  ['this is a test\nof a newline', '', ''], //newline and empty data
  ['this is a "quotation mark" test', '', ''] // quotation test
];

it('correctly encodes data into a CSV', () => {
  expect(papa.unparse(testData)).toMatchSnapshot();
});