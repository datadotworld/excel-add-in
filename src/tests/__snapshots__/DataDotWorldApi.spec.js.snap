// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly encodes data into a CSV 1`] = `
"1,2,3
1,\\"2, 3, 4\\",5
10/12/2012,17%,0.123
\\"this is a test
of a newline\\",,
\\"this is a \\"\\"quotation mark\\"\\" test\\",,"
`;
